ios/AppcuesReactNative.swift

```diff
@@ -88,4 +93,63 @@ class AppcuesReactNative: NSObject {
             resolve(implementation.didHandleURL(url))
         }
     }
+
+    // MARK: Event Emitting
+
+    override func supportedEvents() -> [String]! {
+        [Self.eventName]
+    }
+
+    override func startObserving() {
+        hasListeners = true
+    }
+
+    override func stopObserving() {
+        hasListeners = false
+    }
+}
+
+extension AppcuesReactNative: AppcuesAnalyticsDelegate {
+    func didTrack(analytic: AppcuesAnalytic, value: String?, properties: [String: Any]?, isInternal: Bool) {
+        guard hasListeners else { return }
+
+        let analyticName: String
+        switch analytic {
+        case .event:
+            analyticName = "EVENT"
+        case .screen:
+            analyticName = "SCREEN"
+        case .identify:
+            analyticName = "IDENTIFY"
+        case .group:
+            analyticName = "GROUP"
+        }
+
+        sendEvent(
+            withName: Self.eventName,
+            body: [
+                "analytic": analyticName,
+                "value": value ?? "",
+                "properties": formatProperties(properties),
+                "isInternal": isInternal
+            ])
+    }
+
+    /// Map any supported property types that `sendEvent` doesn't handle by default.
+    private func formatProperties( _ properties: [String: Any]?) -> [String: Any] {
```

Comment:
well this is easier than the Android side 🙃 - just same question about possible array props that might need to be formatted recursively.